bin/docker-entrypoint.sh

```diff
@@ -1,5 +1,11 @@
 #!/bin/sh
 
+# remove the server.pid
+if [ -f $APP_DIR/tmp/pids/server.pid ]; then
```

Comment:
Hrm. What happens if I start things up normally and then also run `docker-compose run --rm web bash`? Won't that remove a valid pid file?  I wonder if we could instead _move_ where the pid file is kept to say `/tmp/server.pid` which would (i think?) get wiped on restart?  At the very least that wouldn't be shared via the host OS.

Thoughts?

Reply:
If I understand properly, this only pertains to situations where a single container is started, stopped, and then resumed later, with the expectation that it will restart the process the normal way. I'm not sure it doesn't violate some of the assumptions docker makes when doing that, so... not sure.

Reply:
I think it's a harmless change as far as I'm concerned, but it might be unexpected for someone who does rely on the existing state being left alone when resuming.

Reply:
I just checked this using the following steps

1. Open two terminal sessions: A and B
2. Make sure A and B are in the project root
3. In **A**: `docker-compose up`
4. In **B**: `docker-compose run web sh`

I find there is no `server.pid` in the container running `sh` in **B**, but this does not interrupt the web process in **A**. 

I think this is because of the way Docker works with containers. It spins up a new container for a run, and re-uses this container if it's not removed with the `--rm` option when spinning it up.

I'm no Docker expert though. So if anyone sees any gotchas I may be missing, please let me know.

Reply:
Yeah, so it appears that while it won't harm the currently running web process it will remove A's tmp/pids/server.pid which feels like the wrong thing to do.

More importantly, I realize that because we mount the directory inside docker, that `tmp/pids/server.pid` is _shared_ between all containers and the host itself.  While I could live with trashing it in the container I don't like the thought of removing the host file.  And honestly it was my mistake to have it write out a pid file there to begin with.  `tmp` shouldn't be shared, but that's a bigger fish to fry.

Let's do this. Instead of removing the file let's have it write the pid file into `/tmp` which won't be shared among multiple containers nor the host.  Something like this?

```
-CMD ["bundle", "exec", "rails", "s", "-b", "0.0.0.0", "-p", "3000"]
+CMD ["bundle", "exec", "rails", "s", "-b", "0.0.0.0", "-p", "3000", "-P", "/tmp/rails_server.pid"]
```

Thoughts?

Reply:
This is new knowledge for me. It looks like it solves the problem.